### DISCUSSION TITLE
Mobileraker companion installation error

### DISCUSSION THREAD

**Discussion by thewolfman56** [Guilouz/Creality-Helper-Script-Wiki#668] [Issues]
I had previously rooted my K1 Max in January of this year and had mobileraker companion installed along with other components while the K1 Max firmware was v1.3.3.5.  I decided to update the script and firmware to get the updated KAMP as I was beginning to get issues with the printer's touchscreen would stop working (was using the stock interface) and would require deleting the bed mesh to allow it to start working again.

I factory reset the printer, upgraded the firmware to v1.3.3.8 (hesitant to try v1.3.3.29 due to issue reports), and installed the helper script.  everything installed normally until I went to install the mobileraker companion.  It started, but after a minute it gave 2 errors (/usr/data/mobileraker_companion/scripts/install.sh: line 194 is missing or something similar (found in my search history)) and crashed the install script.

Any ideas on what I can do to fix this?  I did find the discussion with issues for mobileraker companion and Obico, but I do not use Obico and the only suggestion that looked like it worked was installing it with the newest firmware.

**Comment by MichaelGrenier83** [comment]
Got same problem:
`Checking required system packages are installed...
/usr/data/mobileraker_companion/scripts/install.sh: line 194: opkg: not found
/usr/data/mobileraker_companion/scripts/install.sh: line 195: opkg: not found`

**Reply by JiePieWie** [reply]
I have the same issue on my brand new K1C. Used the script to install a number of packages and ran into the same error when installing Mobileraker. See screenshot below: 
![image](https://github.com/user-attachments/assets/67416f4d-e9ce-46d6-9d29-02807c3b2ff4)
Tried uninstalling this plugin but that also failed with an error and the helper-script crashing/stopping:
![image](https://github.com/user-attachments/assets/9a63f86a-c5b3-43ec-87da-df25e3034cbb)

**Comment by bobcar0** [comment]
I'm now getting the same error on my Ender3v3ke after having previously installed and used it with no problems.

**Comment by tropho23** [comment]
I found a workaround: execute the command `rm -rf /usr/data/mobileraker_companion` to remove the partial files, then attempt install again.  It worked for me on all 12 of my printers.

Another workaround is to power cycle the printer after installing all other modifications, then execute the script and install Mobileraker Companion.  However this isn't possible if you have already tried and failed to install it, and then the above `rm -rf` command is required.

**Reply by bobcar0** [reply]
Thanks for sharing! I'd done a reset so power cycling after mods and before installing did the trick for me.
